pinata/pinata.go

```diff
@@ -0,0 +1,225 @@
+package pinata
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/opensaucerer/bifrost/shared/errors"
+	"github.com/opensaucerer/bifrost/shared/types"
+	"io"
+	"mime/multipart"
+	"net/http"
+	"os"
+	"path/filepath"
+	 "encoding/json"
+)
+
+type MetaData struct {
+    Name   string      `json:"name"`
+}
+
+type SuccessResponse struct {
+	 IpfsHash   string `json:"IpfsHash"`
+	 Timestamp 	string `json:"Timestamp"`
+	 PinSize 		int64 `json:"PinSize"`
+}
+
+/*
+	Upload File to pinata IPFS
+*/
+
+func (p PinataIPFSStorage) UploadFile(path, filename string, options map[string]interface{}) (*types.UploadedFile, error) {
+
+  // validate JWT	
+	success,err := p.TestAuthentication()
```

Comment:
I like this `TestAuthentication` but it should be done at the point of creating the rainbow bridge.

Also, maybe make the name cooler....say `PerformRoundTrip`, `PreFlight` (this one sounds cooler) 🤔